webpack/assets/javascripts/react_app/components/factCharts/index.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Modal, Button, OverlayTrigger, Tooltip } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import helpers from '../../common/helpers';
+import DonutChart from '../common/charts/DonutChart';
+import Loader from '../common/Loader';
+import MessageBox from '../common/MessageBox';
+import { STATUS } from '../../constants';
+import * as FactChartActions from '../../redux/actions/factCharts/';
+
+import { navigateToSearch } from '../../../services/ChartService';
+
+class FactChart extends React.Component {
+  constructor(props) {
+    super(props);
+    helpers.bindMethods(this, ['openModal', 'closeModal']);
+  }
+
+  openModal() {
+    const { showModal, getChartData, data: { id, path, title } } = this.props;
+
+    getChartData(path, id);
+    showModal(id, title);
+  }
+
+  closeModal() {
+    this.props.closeModal();
+  }
+
+  render() {
+    const {
+      factChart,
+      data: { id, title, search },
+      modalToDisplay,
+    } = this.props;
+
+    const handleChartClick =
+      search && search.match(/=$/) ? null : navigateToSearch.bind(null, search);
+
+    const chartProps = {
+      data: factChart.chartData ? factChart.chartData : null,
+      key: `chart-${id}`,
+      onclick: handleChartClick,
+    };
+
+    const chart = <DonutChart {...chartProps} config='large' />;
+
+    const requestErrorMsg =
+      factChart.loaderStatus === STATUS.ERROR
+        ? __('Request Failed')
+        : __('No data available');
+
+    const error = modalToDisplay ? (
+      <MessageBox
+        msg={requestErrorMsg}
+        icontype='error-circle-o'
+        key={`message-${id}`}
+      />
+    ) : (
+      false
+    );
+
+    const tooltip = (
+      <Tooltip id={`viewChartTooltip-${id}`}>{__('Show distribution chart')}</Tooltip>
+    );
+
+    return (
+      <div>
+        <OverlayTrigger placement='top' overlay={tooltip}>
+          <Button onClick={this.openModal}>{__('View Chart')}</Button>
+        </OverlayTrigger>
+        {modalToDisplay && (
+          <Modal show={true} onHide={this.closeModal}>
+            <Modal.Header closeButton>
+              <Modal.Title>
+                <b>
+                  {// eslint-disable-next-line no-undef
+                  Jed.sprintf(__('Fact distribution chart - %s '), title)}
+                </b>
+                <small>
+                  {// eslint-disable-next-line no-undef
+                  Jed.sprintf(
```

Comment:
please add parenthesis around the counter (e.g. like in this screenshot) 
![image](https://user-images.githubusercontent.com/20167/37874300-7a183d4a-3035-11e8-95f7-457fff0595c3.png)

Reply:
added